### PR TITLE
Do schema file parsing after parsing all other files

### LINKED ISSUE #3341: Separate schema file processing and handle after other files
For a number of in-flight projects we need to separate out schema file processing instead of parsing all files from a project at once. This will be used in the new partial parsing and the schema test descriptions.

### PR DESCRIPTION
**Pull request by gshank**
resolves #3341

### Description

Load dictionaries from yaml files and split out schema parsing to enable additional changes in partial parsing and test descriptions.

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by gshank** [comment]
In my branch I have a separate SchemaSourceFile object, but pulling in that would require me to pull in too much of the other work too, so I've just added the 'dict_from_yaml' attribute to SourceFile here, and will fix it up when I merge.

**Review by kwigley — APPROVED** [review]
This looks fine to me knowing that we plan to add back partial parsing functionality. Is there an issue to track that? Do you mind linking that on this PR for posterity?

File: core/dbt/parser/manifest.py [hunk 1]
```diff
@@ -241,13 +255,17 @@ def parse_project(
...
-            for search_key in parser_files[parser_name]:
-                block = FileBlock(self.manifest.files[search_key])
-                self.parse_with_cache(block, parser)
+            for file_id in parser_files[parser_name]:
+                block = FileBlock(self.manifest.files[file_id])
+                if isinstance(parser, SchemaParser):
+                    dct = block.file.dict_from_yaml
+                    parser.parse_file(block, dct=dct)
+                else:
+                    parser.parse_file(block)
```
[the remark this answers is not in the document]
Comment (gshank) [reply core/dbt/parser/manifest.py:264]: Partial parsing (the old way) should still be in place. My big commit doing it the new way is coming up, yes.

[the remark this answers is not in the document]
Comment (gshank) [reply core/dbt/parser/manifest.py:264]: Oops. You're right, it's not calling 'parse_with_cache'. I'm actually surprised that there are no test cases that fail. Putting parse_with_cache back would definitely be more disruptive, so I'll leave it like this. It should only be for a few days.

File: core/dbt/parser/manifest.py [hunk 2]
```diff
@@ -124,6 +129,9 @@ def __init__(
         # State check determines whether the old_manifest and the current
         # manifest match well enough to do partial parsing
         self.manifest.state_check = self.build_manifest_state_check()
+
+        self._perf_info = self.build_perf_info()
```
[the remark this answers is not in the document]
Comment (gshank) [reply core/dbt/parser/manifest.py:133]: Good catch. I've removed it.

**Comment by gshank** [comment]
The ticket number for moving partial parsing is #3217

**Pull request merged**